DependencyInjection/ElaoErrorNotifierExtension.php

```diff
@@ -39,10 +40,17 @@ public function load(array $configs, ContainerBuilder $container)
             $loader = new XmlFileLoader($container, new FileLocator(array(__DIR__ . '/../Resources/config/')));
             $loader->load('services.xml');
 
+            $definition = $container->getDefinition('elao.error_notifier.listener');
+            
             if ($config['mailer'] != 'mailer') {
-                $definition = $container->getDefinition('elao.error_notifier.listener');
                 $definition->replaceArgument(0, new Reference($config['mailer']));
             }
+            
+            if(true === class_exists('Symfony\Component\Console\Event\ConsoleExceptionEvent')) {
```

Comment:
In order to avoid the deprecation, we should favor using the new `console.error` and `ConsoleErrorEvent` event class.
Hence, the condition should be reversed and check the `ConsoleErrorEvent` existence instead, right?

(A space is also missing after the `if`)

Reply:
I thought about that but if we're being 100% BC someone could have extended the Listener with their own class and be expecting it to come through the original method, right?

Reply:
Another would be if they had a listener before it intercepting the error and maybe even halting the code base it'd break that functionality too.

Reply:
You're right, but this bundle throwing a deprecation without any way to fix it isn't right either :/
One in such case would have to update it's event to be registered on `console.error` rather than `console.exception` in order to get rid of the deprecation on their side and to upgrade to 4.x.
So it seems a dead-end. hence I'd change this to get rid of the deprecation on the bundle side at least.

Reply:
But that'd be a BC so are we going to make a BC upgrade/version bump?  In which case wouldn't it make more sense to remove the functionality completely so it will only work on 4.0+ and then leave a 3.X branch?